### ISSUE TITLE
Tests fail

### ISSUE THREAD

**Issue by guimard** [stash/Feersum#26]
Hi,
I'm trying to package Feersum for Debian. For now test fails: see [testers matrix](http://www.cpantesters.org/cpan/report/80a118fe-19a5-11e8-9f72-ac01a5ce2784).

Test done with Debian testing:

1.  test `t/55-psgi-leak.t` never stops

I need to remove "graceful_shutdown" command that never stops

2.  Other errors

```
$ LANG=C make test
"/usr/bin/perl" -MExtUtils::Command::MM -e 'cp_nonempty' -- Feersum.bs blib/arch/auto/Feersum/Feersum.bs 644
"/usr/bin/perl" "-MExtUtils::Command::MM" "-MTest::Harness" "-e" "undef *Test::Harness::Switches; test_harness(0, 'blib/lib', 'blib/arch')" t/*.t
t/01-simple.t ............. ok     
t/02-array-body.t ......... ok     
t/03-env-hash.t ........... ok       
t/04-died.t ............... ok     
t/05-streaming.t .......... ok     
t/06-input.t .............. ok     
t/07-graceful-shutdown.t .. 40/43 
#   Failed test 'SHUTDOWN TOOK TOO LONG'
#   at t/07-graceful-shutdown.t line 88.
# Looks like your test exited with 1 just after 40.
        (in cleanup)    (in cleanup)  at /usr/share/perl5/Test2/API.pm line 173 during global destruction.
t/07-graceful-shutdown.t .. Dubious, test returned 1 (wstat 256, 0x100)
Failed 4/43 subtests 
t/08-read-timeout.t ....... ok     
t/09-magic.t .............. ok     
t/10-respond-304.t ........ ok     
t/11-runner.t ............. ok     
t/12-close-on-drop.t ...... ok     
t/13-pre-fork.t ........... ok     
t/14-guard.t .............. 1/22 
#   Failed test 'guard fired only once'
#   at t/14-guard.t line 92.
#          got: '0'
#     expected: '1'
# Looks like you planned 22 tests but ran 21.
# Looks like you failed 1 test of 21 run.
t/14-guard.t .............. Dubious, test returned 1 (wstat 256, 0x100)
Failed 1/22 subtests 
t/15-write_array.t ........ ok     
t/50-psgi-simple.t ........ ok     
t/51-psgi-streaming.t ..... ok     
t/52-psgi-iohandle.t ...... ok     
t/53-psgi-overloaded.t .... ok     
t/54-psgix-io.t ........... 1/115 close at destruction: Bad file descriptor
close at destruction: Bad file descriptor
close at destruction: Bad file descriptor
close at destruction: Bad file descriptor
close at destruction: Bad file descriptor
close at destruction: Bad file descriptor
close at destruction: Bad file descriptor
close at destruction: Bad file descriptor
t/54-psgix-io.t ........... ok       
t/55-psgi-leak.t .......... ok     
t/60-plack.t .............. ok   
t/61-plack-suite.t ........ ok     
t/62-plack-runner.t ....... 1/6 # found plackup: /usr/bin/plackup (Plack 1.0047)
t/62-plack-runner.t ....... ok   
t/63-plack-apps.t ......... ok   
t/99-critic.t ............. skipped: Author tests not required for installation
t/99-fixme.t .............. skipped: Author tests not required for installation
t/99-manifest.t ........... skipped: Author tests not required for installation
t/99-pod-coverage.t ....... skipped: Author tests not required for installation
t/99-pod.t ................ skipped: Author tests not required for installation

Test Summary Report
-------------------
t/07-graceful-shutdown.t (Wstat: 256 Tests: 40 Failed: 1)
  Failed test:  40
  Non-zero exit status: 1
  Parse errors: Bad plan.  You planned 43 tests but ran 40.
t/14-guard.t            (Wstat: 256 Tests: 22 Failed: 1)
  Failed test:  20
  Non-zero exit status: 1
Files=30, Tests=930, 57 wallclock secs ( 0.23 usr  0.02 sys +  3.13 cusr  0.49 csys =  3.87 CPU)
Result: FAIL
Failed 2/30 test programs. 2/930 subtests failed.
Makefile:1074: recipe for target 'test_dynamic' failed
make: *** [test_dynamic] Error 255
```

**Comment by guimard** [comment]
Also sometimes t/63 test fails:
```
t/63-plack-apps.t ......... 5/6 
#   Failed test at t/63-plack-apps.t line 84.
#          got: '599'
#     expected: '200'

#   Failed test 'found static line (cascade)'
#   at t/63-plack-apps.t line 85.
#                   'Could not read from socket: 'Connection reset by peer'
# '
#     doesn't match '(?^m:^\#\ IS\ THIS\ FILE\ STATICALLY\ SERVED\?$)'
# Looks like you failed 2 tests of 6.
```

**Comment by guimard** [comment]
Ping ?

**Comment by guimard** [comment]
This works (except for t/14) even if it's not well:
```
diff --git a/Feersum.xs b/Feersum.xs
index 4d48c3a..1f9c872 100644
--- a/Feersum.xs
+++ b/Feersum.xs
@@ -2281,7 +2281,7 @@ graceful_shutdown (SV *self, SV *cb)
     ev_io_stop(feersum_ev_loop, &accept_w);
     close(accept_w.fd);
 
-    if (active_conns <= 0) {
+    //if (active_conns <= 0) {
         trace("shutdown is immediate\n");
         dSP;
         ENTER;
@@ -2294,7 +2294,7 @@ graceful_shutdown (SV *self, SV *cb)
         shutdown_cb_cv = NULL;
         FREETMPS;
         LEAVE;
-    }
+    //}
 }
 
 double
```

**Comment by guimard** [comment]
For t/14-guard.t, this Guard object isn't freed (line 68):
```
$w->response_guard(guard {
            $guard_fired++;
            pass "stream writer guard called";
        });
```

**Comment by guimard** [comment]
Hello, all test bugs seem fixed by 1.406, thanks !
